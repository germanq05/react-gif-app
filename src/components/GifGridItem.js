import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ( {title, url} ) => {

    return (
        <div className="card">
            <img src={ url } alt={ url }/>
            <p>{title}</p>
        </div>
    )
}

export default GifGridItem;

GifGridItem.PropTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
