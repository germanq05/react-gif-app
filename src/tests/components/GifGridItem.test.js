import React from 'react';
import GifGridItem from "../../components/GifGridItem";
import { shallow } from 'enzyme';

describe('Pruebas en < GifGridItem />', () => {

    const props = {
        title: "Un titulo",
        url: "https://localhost/algo.jpg"
    };

    const wrapper = shallow( <GifGridItem { ...props }/> );
    
    test('Debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p'); //p porque esta envuelto en un parrafo
        expect( p.text().trim() ).toBe( props.title );
    })

    test('Debe tener la imagen igual al url de los props', () => {
        const img = wrapper.find('img');
        expect( img.props().src ).toBe( props.url );
    })
    
    

})
