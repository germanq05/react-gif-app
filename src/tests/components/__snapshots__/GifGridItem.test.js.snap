// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en < GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="https://localhost/algo.jpg"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
